source/modulo_components/modulo_components/component.py

```diff
@@ -0,0 +1,72 @@
+from threading import Thread
+
+from modulo_components.component_interface import ComponentInterface
+
+
+class Component(ComponentInterface):
+    """
+    Class to represent a Component in python, following the same logic pattern
+    as the c++ modulo_component::Component class.
+    ...
+    Attributes:
+        started (bool): flag that indicates if execution has started or not
+        run_thread (Thread): the execution thread
+    """
+
+    def __init__(self, node_name: str, start_thread=True, *kargs, **kwargs):
+        """
+        Constructs all the necessary attributes and declare all the parameters.
+            Parameters:
+                node_name (str): name of the node to be passed to the base Node class
+                start_thread (bool): start the execution thread at construction
+        """
+        super().__init__(node_name, *kargs, **kwargs)
+        self.__started = False
+        self.__run_thread = None
+        self.add_predicate("in_error_state", False)
```

Comment:
As we discussed just now, this predicate can be added in the base ComponentInterface